src/components/carousel/carousel.ts

```diff
@@ -160,8 +160,9 @@ export default class IgcCarouselComponent extends EventEmitterMixin<
     if (activeSlides.length <= 1) {
       return;
     }
-
-    const idx = this.slides.indexOf(last(added.length ? added : attributes));
+    const idx = this.slides.indexOf(
+      added.length > 0 ? last(added).node : last(attributes)
```

Comment:
Just a minor nit, but this being the same check as in the button group, can those be the same - i.e. `added.length ? ...`